File: IntroToF#/src/CSharp/Lambdas.cs
using System;

namespace CSharp
{
    class Lambdas
    {
        void Sample()
        {
            Action<Func<string>> printMe = x => Console.WriteLine(x());

            printMe(() => "I am totally a lambda!");
            // -----^ I'm not bound to a name so I'm a lambda

            Func<String> alsoALambda = () => "I'm a lambda too!";
            // ------------------------^ I'm also not bound to a name although I am assigned to a variable.

            printMe(alsoALambda);
        }

        string ImDefinitelyNotALambda() { return "I'm not a lambda.... :("; }
        // ----^ I'm bound to a name so I'm not a lambda.
    }
}

File: IntroToF#/src/FSharp/Lambdas.fs
module Lambdas

    open System

    let printMe x = Console.WriteLine(x.ToString())

    let notALambda () = "I'm not a lambda!"
    // -^ I'm bound to a name so I'm not in the lambda club

    printMe notALambda

    printMe (fun () -> "I am totally a lambda!")
    // -----^ I'm not bound to a name so I'm a lambda

    let alsoALambda = (fun () -> "I am totally a lambda too!")
    // ---------------^ I'm still not bound to a name even though I'm assigned to a value

    printMe alsoALambda